src/SFA.DAS.ApplyService.InternalApi/Infrastructure/CompaniesHouseApiClient.cs

```diff
@@ -0,0 +1,135 @@
+
+            var officers = Mapper.Map<IEnumerable<Officer>, IEnumerable<Types.CompaniesHouse.Officer>> (items);
+
+            foreach(var officer in officers)
+            {
+                officer.Disqualifications = await GetOfficerDisqualifications(officer.Id);
+            }
+
+            return officers;
+        }
+
+        private async Task<IEnumerable<Types.CompaniesHouse.Disqualification>> GetOfficerDisqualifications(string officerId)
+        {
+            _logger.LogInformation($"Searching Companies House - Natural Officer's Disqualifications. Officer Id: {officerId}");
+            var apiResponseNatural = await Get<DisqualificationList>($"/disqualified-officers/natural/{officerId}");
+
+            _logger.LogInformation($"Searching Companies House - Corporate Officer's Disqualifications. Officer Id: {officerId}");
+            var apiResponseCorporate = await Get<DisqualificationList>($"/disqualified-officers/corporate/{officerId}");
+
+            var disqualifications = new List<Disqualification>();
+
+            if(apiResponseNatural?.disqualifications != null)
+            {
+                disqualifications.AddRange(apiResponseNatural.disqualifications);
+            }
+            if (apiResponseCorporate?.disqualifications != null)
+            {
+                disqualifications.AddRange(apiResponseCorporate.disqualifications);
+            }
+
+            return Mapper.Map<IEnumerable<Disqualification>, IEnumerable<Types.CompaniesHouse.Disqualification>> (disqualifications);
+        }
+
+        private async Task<IEnumerable<Types.CompaniesHouse.PersonWithSignificantControl>> GetPeopleWithSignificantControl(string companyNumber, bool activeOnly = true)
+        {
+            _logger.LogInformation($"Searching Companies House - People With Significant Control. Company Number: {companyNumber}");
+            var apiResponse = await Get<PersonWithSignificantControlList>($"/company/{companyNumber}/persons-with-significant-control?items_per_page=100");
+
+            var items = activeOnly ? apiResponse.items?.Where(i => i.ceased_on is null) : apiResponse.items;
+            return Mapper.Map<IEnumerable<PersonWithSignificantControl>, IEnumerable<Types.CompaniesHouse.PersonWithSignificantControl>>(items);
+        }
+
+        private async Task<IEnumerable<dynamic>> GetCharges(string companyNumber)
+        {
+            _logger.LogInformation($"Searching Companies House - Charges. Company Number: {companyNumber}");
+            var apiResponse = await Get<ChargeList>($"/company/{companyNumber}/charges");
+            return apiResponse.items;
+            //return Mapper.Map<IEnumerable<Charge>, IEnumerable<Types.CompaniesHouse.Charge>> (apiResponse.items);
+        }
+
+        private async Task<dynamic> GetInsolvencyDetails(string companyNumber)
+        {
+            _logger.LogInformation($"Searching Companies House - Insolvency. Company Number: {companyNumber}");
```

Comment:
this is not being used at the moment but is just a reminder that a mapper would need to be created.